src/app/home-page/components/slideshow/slideshow.component.ts

```diff
@@ -0,0 +1,205 @@
+      translatePostion = -this.translateStep;
+      step = -100;
+      commonCounterAction = 'inc';
+      itemToMove = this.firstItem;
+      itemToMovePosition = this.totalSlidesSize;
+      counterAction = 0;
+    }
+
+    if (this.counter === counter) {
+      this.handleLastAndFirstMoves(step, itemToMove, itemToMovePosition, counterAction);
+    } else {
+      this.commonSlidesMove(translatePostion, commonCounterAction);
+    }
+
+    this.selected = this.counter;
+  }
+
+  private handleClick(arr) {
+    this.subscriptions = arr.map(({ el, callback }) => {
+      const stream = fromEvent(el, 'click');
+      return stream.pipe(throttleTime(400)).subscribe(callback);
+    });
+  }
+
+  private translateItem(item, step: number): void {
+    if (item.nativeElement) {
+      item.nativeElement.style.transform = `translate(${step}%)`;
+    } else {
+      item.style.transform = `translate(${step}%)`;
+    }
+  }
+
+  private moveContainerAndItem(itemToMove: HTMLElement, direction): void {
+    this.translateItem(this.allSlides, this.currentTranslatePosition);
+    this.translateItem(itemToMove, direction);
+  }
+
+  private moveToEdgeSlideWithoutRewind(): void {
+    this.allSlides.nativeElement.style.transition = 'none';
+    let itemToMove;
+    let containerMoveDirection = 0;
+    let itemMoveDirection = 0;
+    let changeTranslatePosition = 0;
+
+    if (this.counter === 0) {
+      itemToMove = this.firstItem;
+    } else {
+      itemToMove = this.lastItem;
+      containerMoveDirection = this.translateStep - 100;
+      changeTranslatePosition = -100 + this.translateStep;
+      itemMoveDirection = 0;
+    }
+
+    this.translateItem(this.allSlides, containerMoveDirection);
+    this.translateItem(itemToMove, itemMoveDirection);
+    this.currentTranslatePosition = changeTranslatePosition;
+
+    this.allSlides.nativeElement.removeEventListener(
+      'transitionend',
+      this.moveToEdgeSlideWithoutRewind
+    );
+  }
+
+  private handleLastAndFirstMoves(changeSlidesPos, itemToMove, positionToMove, counterAction): void {
+    this.currentTranslatePosition = changeSlidesPos;
+    this.moveContainerAndItem(itemToMove, positionToMove);
+    this.allSlides.nativeElement.addEventListener(
+      'transitionend',
+      this.moveToEdgeSlideWithoutRewind
+    );
+    this.counter = counterAction;
+  }
+
+  private commonSlidesMove(translatePostion, counterAction): void {
+    this.currentTranslatePosition += translatePostion;
+    this.translateItem(this.allSlides, this.currentTranslatePosition);
+    counterAction === 'inc' ? this.counter++ : this.counter--;
+  }
+
+  public bulletHandler(i: number): void {
+    this.allSlides.nativeElement.style.transition = '';
+    this.counter = i;
+    this.selected = this.counter;
+    this.currentTranslatePosition = i * -this.translateStep;
+    this.translateItem(this.allSlides, this.currentTranslatePosition);
+  }
+
+  public stopSlideshow(): void {
+    this.continueSlideshow = () => null;
```

Comment:
don't overwrite handler. just set a boolean pause: Boolean

and check in the continue function

Reply:
ty for explained comments, i'll fix it
have nice weekend!

Reply:
done